tests/e2e/api/api.go

```diff
@@ -0,0 +1,158 @@
+		nodeID = node.NodeID
+		uri := node.GetAccessibleURI()
+		client = connectinfopb.NewInfoServiceClient(
+			http.DefaultClient,
+			uri,
+			connect.WithInterceptors(
+				connectclient.SetRouteHeaderInterceptor{Route: "info"},
+			),
+		)
+	})
+
+	ginkgo.It("serves GetNodeVersion", func() {
+		request := connect.NewRequest(&infopb.GetNodeVersionRequest{})
+		response, err := client.GetNodeVersion(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.NotEmpty(response.Msg.Version)
+	})
+
+	ginkgo.It("serves GetNodeID", func() {
+		request := connect.NewRequest(&infopb.GetNodeIDRequest{})
+		response, err := client.GetNodeID(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.Equal(nodeID.String(), response.Msg.NodeId)
+	})
+
+	ginkgo.It("serves GetNetworkID", func() {
+		request := connect.NewRequest(&infopb.GetNetworkIDRequest{})
+		response, err := client.GetNetworkID(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.Equal(network.GetNetworkID(), response.Msg.NetworkId)
+	})
+
+	ginkgo.It("serves GetNetworkName", func() {
+		request := connect.NewRequest(&infopb.GetNetworkNameRequest{})
+		response, err := client.GetNetworkName(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		wantNetworkName := fmt.Sprintf("network-%d", network.GetNetworkID())
+		require.Equal(wantNetworkName, response.Msg.NetworkName)
+	})
+
+	ginkgo.It("serves GetNodeIP", func() {
+		request := connect.NewRequest(&infopb.GetNodeIPRequest{})
+		response, err := client.GetNodeIP(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.Equal(node.StakingAddress.String(), response.Msg.Ip)
+	})
+
+	ginkgo.It("serves GetChainID", func() {
+		request := connect.NewRequest(&infopb.GetChainIDRequest{Alias: "P"})
+		response, err := client.GetChainID(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.Equal(ids.Empty.String(), response.Msg.ChainId)
+	})
+
+	ginkgo.It("serves GetPeers", func() {
+		request := connect.NewRequest(&infopb.GetPeersRequest{})
+		response, err := client.GetPeers(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.NotEmpty(response.Msg.Peers)
+	})
+
+	ginkgo.It("serves GetBootstrapped", func() {
+		request := connect.NewRequest(&infopb.GetBootstrappedRequest{Chain: "P"})
+		response, err := client.GetBootstrapped(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.True(response.Msg.Bootstrapped)
+	})
+
+	ginkgo.It("serves GetUpgrades", func() {
+		request := connect.NewRequest(&infopb.GetUpgradesRequest{})
+		response, err := client.GetUpgrades(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.NotZero(response.Msg)
+	})
+
+	ginkgo.It("serves GetUptime", func() {
+		request := connect.NewRequest(&infopb.GetUptimeRequest{})
+		response, err := client.GetUptime(tc.DefaultContext(), request)
+		require.NoError(err)
+
+		require.NotZero(response.Msg.RewardingStakePercentage)
+		require.NotZero(response.Msg.WeightedAveragePercentage)
```

Comment:
These tests could definitely be better... but it's hard to get something that doesn't flake + doesn't require changes to tmpnet to get better coverage. FWIW, we don't have any api tests right now.

Reply:
Yeah these do benefit from multi-node setups, but for this test specifically we can't guarantee what the exact expected uptime is since we don't know in advance what exact range of time nodes will be up for.